src/content/blog/2022-08-05-new-config-system-part-2.md

```diff
@@ -0,0 +1,512 @@
+            "jsd/check-values": "error"
+        }  
+    }
+];
+```
+
+Here, the plugin is named `jsd` in the config, so the rules also use `jsd` to indicate which plugin they are coming from.
+
+### From `--rulesdir` to runtime plugins
+
+With eslintrc, rules needed to be loaded by the CLI directly in order to be available inside of a config file. This means either bundling custom rules in a plugin or using the `--rulesdir` flag to specify the directory from which ESLint should load custom rules. Both approaches required some extra work to set up and were a frequent cause of frustration for our users.
+
+With flat config, you can load custom rules directly in the config file. Because plugins are now objects directly in the config, you can easily create runtime plugins that exist only in your config file, such as:
+
+```js
+import myrule from "./custom-rules/myrule.js";
+
+export default [
+    {
+        files: ["**/*.js"],
+        plugins: {
+            custom: {
+                rules: {
+                    myrule
+                }
+            }
+        }
+        rules: {
+            "custom/myrule": "error"
+        }  
+    }
+];
+```
+
+Here, a custom rule is imported as `myrule` and then a runtime plugin is created named `custom` to provide that rule to the config as `custom/myrule`. 
+
+As a result, we will be removing `--rulesdir` once the transition to flat config is complete.
+
+### Custom parsers in plugins
+
+One of the weird artifacts of ESLint's development is that parsers were never part of plugins. That's because custom parsers existed long before plugins did, and we never went back to make the two work well together. In flat config, we took the opportunity to fix this problem by allowing plugins to expose parsers in the same that they expose processors and rules. For example, you can now define a plugin that looks like this:
+
+```js
+export default {
+    parsers: {
+        parserName: {
+            parse() { /*... */ }
+        }
+    }
+}
+```
+
+This plugin exposes a parser called `parserName` under the `parsers` key. You can then use this parser in your config like this:
+
+```js
+import custom from "./custom-plugin.js";
+
+export default [
+    {
+        files: ["**/*.js"],
+        plugins: {
+            custom
+        },
+        languageOptions: {
+            parser: "custom/parserName"
+        }
+    }
+];
+```
+
+This config creates a plugin namespace called `custom`. The custom parser can then be accessed using the string `"custom/parserName"`.
+
+### Processors works in a similar way to eslintrc
+
+The `processor` top-level key works mostly the same as in eslintrc, with the primary use case being to use a processor that is defined in a plugin, for example:
+
+```js
+import markdown from "eslint-plugin-markdown";
+
+export default [
+    {
+        files: ["**/*.md"],
+        plugins: {
+            markdown
+        },
+        processor: "markdown/markdown"
+        settings: {
```

Comment:
Is this needed in this example? I feel this could be a little confusing since this section doesn't explain the use of `settings` and that key is explained later.

Reply:
Oh nice catch. Yeah, I’ll remove that.